test/github-task

```diff
@@ -18,137 +18,14 @@
-    status = statuses.get(context, None)
-    current = status.get("description", None)
-    if current and current != expected:
-        sink.status.pop("github", None)
-        sink.status.pop("badge", None)
-        sink.status.pop("irc", None)
-        sys.stderr.write("Verify collision: {0}\n".format(current))
-        return False
-    return True
-
-def rebase(sink):
-    try:
-        sys.stderr.write("Rebasing onto origin/master ...\n")
-        subprocess.check_call([ "git", "fetch", "origin", "master" ])
-        if sink:
-            master = subprocess.check_output([ "git", "rev-parse", "origin/master" ]).strip()
-            sink.status["master"] = master
-        subprocess.check_call([ "git", "rebase", "origin/master" ])
-        return None
-    except:
-        subprocess.call([ "git", "rebase", "--abort" ])
-        traceback.print_exc()
-        return "Rebase failed"
-
-def stop_publishing(sink, ret):
-    def mark_failed():
-        if "github" in sink.status:
-            sink.status["github"]["status"]["state"] = "failure"
-        if 'badge' in sink.status:
-            sink.status['badge']['status'] = "failed"
-        if "irc" in sink.status: # Never send success messages to IRC
-            sink.status["irc"]["channel"] = "#cockpit"
-    def mark_passed():
-        if "github" in sink.status:
-            sink.status["github"]["status"]["state"] = "success"
-        if 'badge' in sink.status:
-            sink.status['badge']['status'] = "passed"
-    if isinstance(ret, basestring):
-        message = ret
-        mark_failed()
-    elif ret == 0:
-        message = "Tests passed"
-        mark_passed()
-    else:
-        message = "{0} tests failed".format(ret)
-        mark_failed()
-    sink.status["message"] = message
-    if "github" in sink.status:
-        sink.status["github"]["status"]["description"] = message
-    del sink.status["extras"]
-    sink.flush()
-
-def eintr_retry_call(func, *args):
-    while True:
-        try:
-            return func(*args)
-        except (OSError, IOError) as e:
-            if e.errno == errno.EINTR:
-                continue
-            raise
-
-# The goal here is that after 60 seconds we call check_publishing
-def wait_testing(proc, sink, github):
-    count = 0
-    flags = os.WNOHANG
-    while True:
-        pid, status = eintr_retry_call(os.waitpid, proc.pid, flags)
-        if count < 60:
-            time.sleep(1)
-            count += 1
-        elif count == 60:
-            if sink and not check_publishing(sink, github):
-                try:
-                    proc.terminate()
-                except OSError:
-                    pass
-            flags = 0
-        if pid == proc.pid:
-            if os.WIFSIGNALED(status):
-                return os.WTERMSIG(status)
-            else:
-                return os.WEXITSTATUS(status)
+import testpulltask
 
 def main():
     parser = argparse.ArgumentParser(description='Perform next testing task from Github')
```

Comment:
task relevant options could be populated by a function in the task object, since they're passed through anyway

I'm not sure if this makes sense or makes everything too complicated.